src/backend/distributed/commands/utility_hook.c

```diff
@@ -912,6 +912,20 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 		{
 			CommitTransactionCommand();
 			StartTransactionCommand();
+
+			/*
+			 * Even if we started a new transaction, now we still hold
+			 * ExclusiveLock (acquired by VirtualXactLockTableInsert) on current
```

Comment:
```
 * ExclusiveLock (acquired by VirtualXactLockTableInsert) 
 * ```
This doesn't sounds accurate, `VirtualXactLockTableInsert` doesn't hold any exclusive locks?

I'm actually not sure the effects of `VirtualXactLockTableCleanup ` at this point, still trying to understand it a little more

Reply:
>This doesn't sounds accurate, VirtualXactLockTableInsert doesn't hold any exclusive locks?

As far as I saw when debugging, lock on vxid is acquired right after `VirtualXactLockTableInsert` is executed